### ISSUE TITLE
Question about instance loss calculation

### ISSUE THREAD

**Issue by mvish7** [koyeongmin/PINet_new#23]
Hi,
Thanks for making this great work public. 

I have a question regarding the calculation of instance loss for PINet of CULane dataset. could you please explain what exactly is happening in the following lines?

```
compute loss for similarity

feature_map = feature.view(real_batch_size, self.p.feature_size, 1, self.p.grid_y*self.p.grid_x)
feature_map = feature_map.expand(real_batch_size, self.p.feature_size, self.p.grid_y*self.p.grid_x, 
                                                          self.p.grid_y*self.p.grid_x)#.detach()

point_feature = feature.view(real_batch_size, self.p.feature_size, self.p.grid_y*self.p.grid_x,1)
point_feature = point_feature.expand(real_batch_size, self.p.feature_size, self.p.grid_y*self.p.grid_x, 
                                                            self.p.grid_y*self.p.grid_x)#.detach()

distance_map = (feature_map-point_feature)**2 
distance_map = torch.sum( distance_map, dim=1 ).view(real_batch_size, 1, self.p.grid_y*self.p.grid_x, 
                                             self.p.grid_y*self.p.grid_x)
```

As per my understanding (please correct me if I'm wrong):

The feature_map and point_feature are derived from the output of the feature branch of PINet by doing some resizing and expanding operations. and then these feature_map and point_feature are used to calculate the distance_map.

My exact question is:

what is the difference between feature_map and point_feature?? and what is the significance of feature_map and point_feature?

Thanks :)

**Comment by koyeongmin** [comment]
feature_map and point_feature have different forms.
feature_map: feature.view(real_batch_size, self.p.feature_size, 1, self.p.grid_y*self.p.grid_x)
point_feature: feature.view(real_batch_size, self.p.feature_size, self.p.grid_y*self.p.grid_x,1)
point_feature has transposed form of feature_map.
Thank you!

**Comment by mvish7** [comment]
Hey,
Thanks for the clarification. I understood why it is being that way once I debugged the code
